diff --git a/node_modules/async-promise-queue/index.js b/node_modules/async-promise-queue/index.js
index 962c645..baf0f47 100644
--- a/node_modules/async-promise-queue/index.js
+++ b/node_modules/async-promise-queue/index.js
@@ -10,8 +10,8 @@ module.exports = function queue(worker, work, concurrency) {
     } else {
       var q = require('async').queue(worker, concurrency);
       var firstError;
-      q.drain = resolve;
-      q.error = function(error) {
+      q.drain(resolve);
+      q.error(function(error) {
         if (firstError === undefined) {
           // only reject with the first error;
           firstError = error;
@@ -21,11 +21,11 @@ module.exports = function queue(worker, work, concurrency) {
         q.kill();
 
         // but wait until all pending work completes before reporting
-        q.drain = function() {
+        q.drain(function() {
           reject(firstError);
-        };
+        });
 
-      };
+      });
       q.push(work);
     }
   });
